gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -463,6 +469,18 @@ public void apply(JobListener jobListener, JobContext jobContext)
           return;
         }
 
+        // calculation of total bytes to copy in a job used to track a job's copy progress
+        if (jobState.getPropAsBoolean(ConfigurationKeys.REPORT_JOB_PROGRESS, ConfigurationKeys.DEFAULT_REPORT_JOB_PROGRESS)) {
+          if (workUnitStream.isSafeToMaterialize()) {
+            long totalSizeInBytes = sumWorkUnitsSizes(workUnitStream);
+            this.jobContext.getJobState().setProp(TOTAL_BYTES_TO_COPY, totalSizeInBytes);
```

Comment:
TOTAL_BYTES_TO_COPY seems very Distcp-centric. We may want to think more broadly and just call this TOTAL_WORK_UNIT_SIZE. In case when individual sources do not provide workunit sizes, this should sum to total number of workunits.